src/rgw/rgw_auth_s3.cc

```diff
@@ -700,13 +703,18 @@ std::string get_v4_string_to_sign(CephContext* const cct,
                                   const sha256_digest_t& canonreq_hash)
 {
   const auto hexed_cr_hash = buf_to_hex(canonreq_hash);
-  const auto string_to_sign = std::string()
+
+  /* We aren't adding the extra byte for the terminating 0 as it would be
+   * annihilated by the "-1" coming from hexed_cr_hash. */
```

Comment:
Thanks for pointing this out and for sharing the knowledge! :-) I've just pushed the amended commit. Also a trivial merge conflict in `rgw_op.cc` has been resolved.